pkg/cmd/pulumi/new.go

```diff
@@ -1112,3 +1117,34 @@ func containsWhiteSpace(value string) bool {
 	}
 	return false
 }
+
+// validateStackProjectName takes a stack name and a project name and returns an error if they are not the same.
+//   - projectName comes from the --name flag.
+//   - stackName comes from the --stack flag. stackName can be a stack name or a fully qualified stack reference
+//     i.e org/project/stack
+func validateStackProjectName(b backend.Backend, stackName, projectName string) error {
+	if stackName == "" || projectName == "" {
+		// No potential for conflicting project names.
+		return nil
+	}
+
+	// Catch the case where the user has specified a fully qualified stack reference.
+	if strings.Count(stackName, "/") == 2 {
```

Comment:
No need to check for if there's two "/" here, just let the backend do the parse.